test/integration/targets/ansible/test_version.py

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
```

Comment:
Instead of using a script that subprocesses ansible, you can add a test to `test/units/cli/test_adhoc.py` that utilizes the `capsys` pytest fixture to get the output: https://docs.pytest.org/en/latest/capture.html

Reply:
Nice! I will check that out.

Reply:
I have this mostly working. The only issue I have when running using `AdHocCLI` is the `%(prog)s` is not `ansible` but a `pytest` string:

```
test/units/cli/test_adhoc.py pytest 2.9.0.dev0
  config file = /Users/sdoran/.ansible.cfg
  configured module search path = ['/Users/sdoran/Source/ansible/library']
  ansible python module location = /Users/sdoran/Source/ansible/lib/ansible
  executable location = /Users/sdoran/.pyenv/versions/3.7.3/bin/pytest
  python version = 3.7.3 (default, Apr 18 2019, 13:51:13) [Clang 10.0.1 (clang-1001.0.46.3)]
```

Is there any way to fix that? Because I think testing the content of that first line is the most critical part.

Reply:
There are likely a few ways to address that.  

Looking at how you are calling the `version` function:

```
ansible_version = to_native(version(getattr(parser, 'prog')))
```

That is using `prog` from `parser`.  `parser` is available at `adhoc_cli.parser` so you can likely just monkeypatch that.

Otherwise, maybe monkeypatch `sys.argv[0]`.

Reply:
I wasn't having any luck patching the `parser` object, but patching `sys.argv` did work.